### PR TITLE
[DEM] Some checks added to the DEM constitutive laws

### PR DESCRIPTION
**Pull request by maceligueta**
These checks should help detecting anomalies in the input files.

### CODE REVIEW

File: applications/DEMApplication/custom_constitutive/DEM_continuum_constitutive_law.cpp [hunk 1]
```diff
@@ -24,6 +24,40 @@ namespace Kratos {
     void DEMContinuumConstitutiveLaw::SetConstitutiveLawInProperties(Properties::Pointer pProp, bool verbose) const {
         if(verbose) KRATOS_INFO("DEM") << "Assigning DEMContinuumConstitutiveLaw to Properties " << pProp->Id() << std::endl;
         pProp->SetValue(DEM_CONTINUUM_CONSTITUTIVE_LAW_POINTER, this->Clone());
+        this->Check(pProp);
+    }
+
+    void DEMContinuumConstitutiveLaw::Check(Properties::Pointer pProp) const {
```
Comment (farrufat-cimne) [inline applications/DEMApplication/custom_constitutive/DEM_continuum_constitutive_law.cpp:30]: When using continuum constitutive models, a discontinuum constitutive law is also always defined in the mdpa. Wouldn't it be redundant to define a check with the same variables already being checked in the  DEMDiscontinuumConstitutiveLaw::Check  ?

File: applications/DEMApplication/test_examples/DEM_run_all_benchmarks_grid.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 if "OMP_NUM_THREADS" in os.environ:
     max_available_threads = int(os.environ['OMP_NUM_THREADS'])
 else:
-    max_available_threads = mp.cpu_count() - 1
+    max_available_threads = mp.cpu_count()
```
Comment (farrufat-cimne) [inline applications/DEMApplication/test_examples/DEM_run_all_benchmarks_grid.py]: I wouldn't recommend using all the available cpu's when running the benchmarks as grid. Leaving 1 free cpu allows the user to keep working and using other applications and system processes.

File: applications/DEMApplication/custom_constitutive/DEM_continuum_constitutive_law.cpp (hunk 1 above)
Comment (maceligueta) [reply applications/DEMApplication/custom_constitutive/DEM_continuum_constitutive_law.cpp:30]: You are right that we are repeating code here, but currently, the continuum and discontinuum cl's are completely separate objects. I thought about linking them somehow, but we should discuss it carefully. Deriving one from the other is an option, but we might loose some flexibility when selecting them from the GUI. It is feasible, though. For the moment I treated them totally separately.

File: applications/DEMApplication/test_examples/DEM_run_all_benchmarks_grid.py (hunk 2 above)
Comment (maceligueta) [reply applications/DEMApplication/test_examples/DEM_run_all_benchmarks_grid.py]: Ok. I put '-1' back, but with an extra check that the number of processors is not 0. Thanks @farrufat-cimne !

Comment (farrufat-cimne) [reply applications/DEMApplication/test_examples/DEM_run_all_benchmarks_grid.py]: true!

**Comment by maceligueta** [comment]
Can this be approved?

**Review by salvalatorre — APPROVED** [review]
Looks fine...

**Pull request merged**